packages/story-editor/src/app/media/utils/useProcessMedia.js

```diff
@@ -168,6 +172,91 @@ function useProcessMedia({
     ]
   );
 
+  const trimExistingVideo = useCallback(
+    ({ resource: oldResource, start, end }) => {
+      const { src: url, mimeType } = oldResource;
+
+      const trim = {
+        original: oldResource.id,
+        start,
+        end,
+      };
+
+      const onUploadStart = () => {
+        updateExistingElements({
+          oldResource: {
+            ...oldResource,
+            trim,
+            isTrimming: true,
+          },
+        });
+      };
+
+      const onUploadError = () => {
+        updateExistingElements({
+          oldResource: { ...oldResource, isTrimming: false },
+        });
+      };
+
+      const onUploadSuccess = ({ resource }) => {
+        copyResourceData({ oldResource, resource });
+        updateOldMutedObject(oldResource.id, resource.id);
+        if ('video' === resource.type && !resource.local) {
+          if (!resource.posterId) {
+            uploadVideoPoster(resource.id, resource.src);
+          }
+          if (resource.isMuted === null) {
+            updateVideoIsMuted(resource.id, resource.src);
+          }
+        }
+      };
+
+      const onUploadProgress = ({ resource }) => {
+        const oldResourceWithId = { ...resource, id: oldResource.id };
+        updateExistingElements({
+          oldResource: oldResourceWithId,
+        });
+      };
+
+      const process = async () => {
+        let file = false;
+        try {
+          file = await fetchRemoteFile(url, mimeType);
+        } catch (e) {
+          // Ignore for now.
+          return;
+        }
+
+        await uploadMedia([file], {
+          onUploadSuccess,
```

Comment:
Should we copy over the existing poster for the trimmed video or should force generate a new one @swissspidy ?

Reply:
What is they already had a custom poster on it?

Reply:
Thinking about this more, let's just copy over the existing poster. It's the safer approach as the generation could fail. If users are not happy with the poster they can easily remove it to have a new one generated.

Reply:
It now copies the existing poster.